bumble/drivers/intel.py

```diff
@@ -0,0 +1,343 @@
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Support for Intel AX210 USB dongle.
+Based on the Linux kernel implementation.
+(see `drivers/bluetooth/btintel.c`)
+"""
+
+# -----------------------------------------------------------------------------
+# Imports
+# -----------------------------------------------------------------------------
+import logging
+from dataclasses import dataclass
+from bumble.hci import hci_vendor_command_op_code, HCI_Command, STATUS_SPEC, HCI_SUCCESS
+from typing import Optional, Dict
+import asyncio
+import sys
+
+# -----------------------------------------------------------------------------
+# Logging
+# -----------------------------------------------------------------------------
+logger = logging.getLogger(__name__)
+
+# -----------------------------------------------------------------------------
+# Constants
+# -----------------------------------------------------------------------------
+CSS_HEADER_OFFSET = 8
+ECDSA_OFFSET = 644
+OPERATIONAL_FW = 0x03
+INTEL_VENDOR_ID = 0x0000
+HW_PLATFORM = 0x37
+MAX_FRAGMENT_PAYLOAD = 252
+RSA_HEADER_LEN = 644
+ECDSA_HEADER_LEN = 320
+
+LINUX_FW_PATH = "/lib/firmware/intel/"
+
+# -----------------------------------------------------------------------------
+# HCI Commands
+# -----------------------------------------------------------------------------
+HCI_INTEL_READ_VERSION_COMMAND = hci_vendor_command_op_code(0xFC05)  # type: ignore
+HCI_INTEL_SECURE_SEND_COMMAND = hci_vendor_command_op_code(0xFC09)  # type: ignore
+HCI_INTEL_RESET_COMMAND = hci_vendor_command_op_code(0xFC01)  # type: ignore
+HCI_Command.register_commands(globals())
+
+
+@HCI_Command.command(  # type: ignore
+    fields=[("param", 1)],
+    return_parameters_fields=[
+        ("status", STATUS_SPEC),
+        ("data", "*"),
+    ],
+)
+class HCI_Intel_Read_Version_Command(HCI_Command):
+    pass
+
+
+@HCI_Command.command(  # type: ignore
+    # fields=[("param", MAX_FRAGMENT_PAYLOAD + 1)],
+    fields=[("param", "*")],
+    return_parameters_fields=[
+        ("data", "*"),
+    ],
+)
+class Hci_Intel_Secure_Send_Command(HCI_Command):
+    pass
+
+
+@HCI_Command.command(  # type: ignore
+    fields=[
+        ("reset_type", 1),
+        ("patch_enable", 1),
+        ("ddc_reload", 1),
+        ("boot_option", 1),
+        ("boot_param", 4),
+    ],
+    return_parameters_fields=[
+        ("data", "*"),
+    ],
+)
+class Hci_Intel_Reset_Command(HCI_Command):
+    pass
+
+
+# -----------------------------------------------------------------------------
+
+
+@dataclass
+class IntelTLV:
+    type: int
+    lenght: int
+    value: bytearray
+
+
+@dataclass
+class IntelVersionTlv:
```

Comment:
To be consistent, either this class should be named `IntelVersionTLV`, or `IntelTLV` should be `IntelTlv`